src/admin/UserList.tsx

```diff
@@ -1,27 +1,77 @@
-import React, { useMemo, useReducer, useRef, useState } from "react";
+import React, { useRef, useState, useEffect } from "react";
 import { getApiUrl } from "../api/api";
 import { Modal, Button } from "react-bootstrap";
 
 import { User } from "../shared/types";
-import ProfileForm from "../profile/ProfileForm";
+import ProfileForm, { useProfileForm } from "../profile/ProfileForm";
 
 import "./UserList.scss";
 import { useAuth } from "../auth/AuthProvider";
 import ERRORS from "../shared/errors";
-import { RichTable, Column, RichTableBag } from "../shared/components/RichTable";
+import {
+  RichTable,
+  Column,
+  RichTableBag,
+} from "../shared/components/RichTable";
 
-interface State {
-  allUsers: User[];
-  filteredUsers: User[] | undefined;
-  currentUser: User | undefined;
-  showEditUser: boolean;
-  showCreateUser: boolean;
+interface UserEditModalProps {
+  user?: User;
+  show: boolean;
+  setShow: (show: boolean) => void;
+  refreshTable: () => void;
 }
 
-type Action =
-  { type: 'set edit user', data: User | undefined } |
-  { type: 'show edit user', data: boolean } |
-  { type: 'show create user', data: boolean };
+const UserEditModal: React.FC<UserEditModalProps> = (
+  props: UserEditModalProps
+) => {
+  const context = useProfileForm(props.user);
+
+  // grab a reference to the form state here, or else it won't update properly
+  const formState = context.formState;
+
+  return (
+    <Modal
+      show={props.show}
+      onHide={() => {
+        props.setShow(false);
+      }}
+      size="lg"
+    >
+      <Modal.Header>
+        <Modal.Title>{props.user ? "Edit User" : "Create User"}</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <ProfileForm
+          context={context}
+          formId="editUserForm"
+          user={props.user}
+          onSubmit={async () => {
+            props.refreshTable();
+            props.setShow(false);
```

Comment:
None of these should run if the submission was unsuccessful.

As well, I'd advise that the modal stay open even on a successful submit, so that we don't have to keep reopening it when creating multiple users.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/1fd4c2e2d1cc3cd580cf518d1928501bee99dc2d

To make that use case work, I also reset the form when a create completes. But it feels kind of... weird since it tells you that you created a user whose data is no longer there. Let me know what you think.

Reply:
The ideal behaviour here would be:

- User fills in info, and submits
- Submission is successful
- Form clears, and an alert states that the previous submission was successful
- repeat

The key is that a notification is given on success. Seeing as that's been implemented, I think this is fine.